apps/dot_ext/views/authorization.py

```diff
@@ -35,6 +44,8 @@ def dispatch(self, request, *args, **kwargs):
             # Create new authorization flow trace UUID in session and AuthFlowUuid instance, if subclass is not ApprovalView
             create_session_auth_flow_trace(request)
 
+        validate_app_is_active(request)
+
         return super().dispatch(request, *args, **kwargs)
```

Comment:
@JFU-GIT A 500 result code is showing here vs. the intended 403.

I believe that the `OAuthToolkitError` class from `oauth2_provider.exceptions` needs to be used here to generate a related response to return. See the `form_valid()` method for an example.

FYI: I've been playing with this, but not getting it to work yet :-)

Reply:
applied changes and fixed tests.